src/components/KInput/KInput.vue

```diff
@@ -319,196 +238,148 @@ export default {
-        }
+    &.has-after-content {
+      .k-input {
+        padding-right: calc($kInputPaddingX * 2 + $kInputIconSize);
       }
     }
   }
 }
 
-.help {
+.k-input {
+  background-color: var(--kui-color-background, $kui-color-background);
+  border: 0;
+  border-radius: var(--kui-border-radius-30, $kui-border-radius-30);
+  box-shadow: var(--kui-shadow-border, $kui-shadow-border);
   color: var(--kui-color-text, $kui-color-text);
-  display: block;
+  font-family: var(--kui-font-family-text, $kui-font-family-text);
   font-size: var(--kui-font-size-30, $kui-font-size-30);
-  margin: var(--kui-space-40, $kui-space-40) var(--kui-space-0, $kui-space-0) var(--kui-space-0, $kui-space-0);
-}
-
-.input-icon {
-  align-items: center;
-  display: inline-flex;
-  pointer-events: none;
-  position: absolute;
-
-  &.clickable {
-    cursor: pointer;
-    pointer-events: auto;
-  }
-}
-
-.has-error {
-  color: var(--kui-color-text-danger, $kui-color-text-danger);
-  font-weight: var(--kui-font-weight-medium, $kui-font-weight-medium);
-}
-
-.k-input-wrapper {
-  position: relative;
-
-  input.k-input {
-    -webkit-appearance: none;
-    appearance: none;
-  }
-
-  & .k-input-label-wrapper-large .has-error,
-  & .k-input-large + .has-error {
-    font-size: var(--kui-font-size-20, $kui-font-size-20);
-    line-height: var(--kui-line-height-20, $kui-line-height-20);
-    margin-top: var(--kui-space-20, $kui-space-20);
+  font-weight: var(--kui-font-weight-regular, $kui-font-weight-regular);
+  line-height: var(--kui-line-height-40, $kui-line-height-40);
+  outline: none;
+  padding: var(--kui-space-40, $kui-space-40) $kInputPaddingX;
+  transition: box-shadow $kongponentsTransitionDurationTimingFunction;
+  width: 100%;
+
+  &::placeholder {
+    color: var(--kui-color-text-neutral, $kui-color-text-neutral);
   }
 
-  & .k-input-label-wrapper-medium .has-error,
-  & .k-input-medium + .has-error {
-    font-size: var(--kui-font-size-10, $kui-font-size-10);
-    line-height: var(--kui-line-height-10, $kui-line-height-10);
-    margin-top: var(--kui-space-10, $kui-space-10);
+  &:hover {
+    box-shadow: var(--kui-shadow-border-primary-weak, $kui-shadow-border-primary-weak);
   }
 
-  & .k-input-label-wrapper-small .has-error,
-  & .k-input-small + .has-error {
-    font-size: var(--kui-font-size-10, $kui-font-size-10);
-    line-height: var(--kui-line-height-10, $kui-line-height-10);
-    margin-top: var(--kui-space-10, $kui-space-10);
+  &:focus {
+    box-shadow: var(--kui-shadow-border-primary, $kui-shadow-border-primary), var(--kui-shadow-focus, $kui-shadow-focus);
   }
 
-  .text-on-input {
-    @include overlay-label-input;
+  &:disabled {
+    @include kInputDisabledRadOnly;
 
-    label:not(.disabled):not(.readonly).hovered,
-    label:not(.disabled):not(.readonly):hover {
-      color: var(--kui-color-text-primary, $kui-color-text-primary);
-    }
+    // need this to ensure user select is disabled
+    pointer-events: none;
```

Comment:
question: I don't think we want to disallow pointer events when disabled; what's the reason for this?